### PR TITLE
072324.02/first release approval button

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Reorder Models into Individual Files
- Get SPA Client Test to All green State
- Enable Testing on Pipeline only for client SPA
- Fix First Release Approval Bug so that button state is correct based on approval status and configured preferred approvers

### CODE REVIEW

File: src/dotnet/APIView/apiview.yml [hunk 1]
```diff
@@ -197,122 +197,141 @@ extends:
...
+              - script: |
+                  npm install
+                workingDirectory: $(SPAClientProjectDirectory)
+                displayName: "Install Client SPA Packages"
+
+              - script: |
+                  ng test --no-watch --no-progress --browsers=ChromeHeadless
+                workingDirectory: $(SPAClientProjectDirectory)
+                displayName: "Run SPA Tests"
+              #- template: /eng/common/pipelines/templates/steps/cosmos-emulator.yml
```
Comment (maririos) [inline src/dotnet/APIView/apiview.yml:226]: is the idea later to remove this code?

File: src/dotnet/APIView/ClientSPA/src/app/_components/review-page-options/review-page-options.component.spec.ts [hunk 2]
```diff
@@ -18,4 +53,37 @@ describe('ReviewPageOptionsComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  describe('First Release Approval Button', () => {
```
Comment (maririos) [inline src/dotnet/APIView/ClientSPA/src/app/_components/review-page-options/review-page-options.component.spec.ts:57]: nice! I see the tests for your changes here! pretty cool

File: eng/pipelines/templates/steps/apiview-ui-tests.yml [hunk 3]
```diff
@@ -92,7 +92,7 @@ steps:
...
     "APIVIEW_API_KEY": "$(azuresdk-apiview-apikey)"
 
-- task: PublishBuildArtifacts@1
+- task: 1ES.PublishPipelineArtifact@1
   inputs:
-    pathtoPublish: '$(Build.SourcesDirectory)\src\dotnet\APIView\APIViewWeb\Client\playwright-report'
-    artifactName: 'Client-Side Test Reports'
\ No newline at end of file
+    path: '$(Build.SourcesDirectory)\src\dotnet\APIView\APIViewWeb\Client\playwright-report'
+    artifact: 'Client-Side Test Reports'
```
Comment (maririos) [inline eng/pipelines/templates/steps/apiview-ui-tests.yml:98]: is this the report of the tests? could you point me to where I could find it?

**Review by maririos — APPROVED** [review]

File: src/dotnet/APIView/apiview.yml (hunk 1 above)
Comment (chidozieononiwu) [reply src/dotnet/APIView/apiview.yml:226]: Testing has been disabled for a while. I think it was during the review revisions restructure. II will reenable it as part of another change. But some of the tests are no longer required now.

File: eng/pipelines/templates/steps/apiview-ui-tests.yml (hunk 3 above)
Comment (chidozieononiwu) [reply eng/pipelines/templates/steps/apiview-ui-tests.yml:98]: I don't yet have reporting for the tests. But it reports on the pipeline log and will break if a test fails.

Comment (maririos) [reply eng/pipelines/templates/steps/apiview-ui-tests.yml:98]: great. if we can see which test breaks then we should be good

**Pull request merged**